src/app/pages/rpc-stx-transfer-stx/use-rpc-stx-transfer-stx.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+
+import { createRequestEncoder, stxTransferStx } from '@leather.io/rpc';
+import { type StacksUnsignedTokenTransferOptions, TransactionTypes } from '@leather.io/stacks';
+import { createMoney, isDefined } from '@leather.io/utils';
+
+import { initialSearchParams } from '@app/common/initial-search-params';
+import { useBreakOnNonCompliantEntity } from '@app/query/common/compliance-checker/compliance-checker.query';
+import { useNextNonce } from '@app/query/stacks/nonce/account-nonces.hooks';
+import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentStacksNetworkState } from '@app/store/networks/networks.hooks';
+
+const { decode } = createRequestEncoder(stxTransferStx.request);
+
+function useTransferStxRequest() {
+  const request = initialSearchParams.get('rpcRequest');
+  if (!request) throw new Error('Missing rpcRequest');
+  return { request: decode(request) };
+}
+
+export function useRpcStxTransferStxTxOptions(): StacksUnsignedTokenTransferOptions {
+  const { request } = useTransferStxRequest();
+  const account = useCurrentStacksAccount();
+  const { data: nextNonce } = useNextNonce(account?.address ?? '');
+  const network = useCurrentStacksNetworkState();
+
+  useBreakOnNonCompliantEntity(
+    [request.params.address ?? account?.address, request.params.recipient].filter(isDefined)
+  );
+
+  return useMemo(
+    () => ({
+      txType: TransactionTypes.StxTokenTransfer,
+      amount: createMoney(request.params.amount, 'STX'),
+      fee: createMoney(request.params.fee ?? 0, 'STX'),
+      memo: request.params.memo ?? '',
+      recipient: request.params.recipient,
+      network,
+      nonce: request.params.nonce ?? nextNonce?.nonce ?? '',
```

Comment:
Isn't the `NonceLoader` component here to prevent logic like this being repeated?

Reply:
Yeah, good point, I'll remove.